core/src/main/java/com/linecorp/armeria/internal/common/stream/FixedStreamMessage.java

```diff
@@ -336,10 +359,19 @@ private void abort1(Throwable cause, boolean subscribed) {
         if (completed) {
             return;
         }
+        completed = true;
```

Comment:
Doesn't have to be handled in this PR:

nit; I think it would be easier to reason about if this path was only called from the event loop.

I understand that functionally there is no difference when `subscribed = false` since `completed` isn't used and the targets to clean up is null anyways.
Having said this, I'm wondering if it's just easier to set `completionFuture` when the executor is set during abortion:

```
        if (executorUpdater.compareAndSet(this, null, ImmediateEventExecutor.INSTANCE)) {
            // No subscription was made.
            completionFuture.completeExceptionally(finalCause);
        } else {
```

Reply:
>  Having said this, I'm wondering if it's just easier to set completionFuture when the executor is set during abortion:

Did you think of inlining the code? For example:
```java
if (executorUpdater.compareAndSet(this, null, ImmediateEventExecutor.INSTANCE)) {
    cleanupObjects(cause);
    completionFuture.completeExceptionally(cause);
}
```

Reply:
> Did you think of inlining the code? For example:

I missed that we need to still call `cleanupObjects`, but yes I think this is what I thought

Reply:
setting `completed` to true could be useful to prevent double abortion although the result without `completed` is the same.

Reply:
> prevent double abortion

I see, I missed this intention. I think it's fine to leave as-is then 👍